sbtc/src/leb128.rs

```diff
@@ -0,0 +1,556 @@
+            test_value(value, 9).unwrap();
+        }
+
+        /// Tests that 10-byte values round-trip correctly
+        #[test]
+        fn test_10_byte_values(value in VALUE_MAX_9_BYTES+1..=VALUE_MAX_10_BYTES) {
+            test_value(value, 10).unwrap();
+        }
+
+        /// Tests bit-level encoding structure
+        ///
+        /// Verifies the internal structure of encoded values:
+        /// - Correct continuation bit usage
+        /// - Proper bit positioning in payload
+        /// - Value reconstruction from encoded bytes
+        #[test]
+        fn test_bit_encoding_structure(value: u64) {
+            let mut bytes = Vec::new();
+            Leb128::encode_into(value, &mut bytes);
+
+            // Encoding size should never exceed 10 bytes (for u64 maximum)
+            prop_assert!(bytes.len() <= 10,
+                "Encoding should be at most 10 bytes for u64");
+
+            // Reconstruct value manually to verify encoding integrity
+            let mut reconstructed = 0u64;
+            for (i, &byte) in bytes.iter().enumerate() {
+                reconstructed |= ((byte & LOWER_BITS_MASK) as u64) << (i * 7);
+            }
+
+            prop_assert_eq!(reconstructed, value,
+                "Manual reconstruction from bytes failed");
+        }
+
+        /// Tests sequential reading of multiple values
+        ///
+        /// Validates cursor-based reading essential for decoding segments:
+        /// - Position tracking between values
+        /// - Sequential value extraction
+        /// - Handling of heterogeneous value sizes
+        #[test]
+        fn test_cursor_reading(values in proptest::collection::vec(any::<u64>(), 1..100)) {
+            let mut encoded = Vec::new();
+
+            // Encode multiple values sequentially
+            for &value in &values {
+                Leb128::encode_into(value, &mut encoded);
+            }
+
+            // Read back using cursor
+            let mut cursor = Cursor::new(encoded.as_slice());
+            let mut decoded = Vec::new();
+
+            while cursor.position() < encoded.len() as u64 {
+                let value = cursor.read_leb128().unwrap();
+                decoded.push(value);
+            }
+
+            // Verify all values were decoded correctly
+            prop_assert_eq!(values, decoded);
+        }
+
+        /// Tests encoding at byte size transitions
+        ///
+        /// Verifies correct byte count at each size boundary:
+        /// - Maximum values for each byte count
+        /// - Minimum values requiring additional bytes
+        /// - Consistent encoding across all boundaries
+        #[test]
+        fn test_all_compression_boundaries(bytes in 1u32..MAX_BYTES as u32) {
+            // Get boundary values
+            let max_value = max_value_for_bytes(bytes);
+            let min_next_value = if bytes < 10 { min_value_for_next_bytes(bytes) } else { u64::MAX };
```

Comment:
nit: can't we replace `min_value_for_next_bytes` with `max_value + 1`? (except for 10 bytes)

Reply:
Yeah, that method was a leftover [4e637a4](https://github.com/stacks-network/sbtc/pull/1523/commits/4e637a45ef835b0d3ca871e44380875a0918b8ad)